### PR TITLE
[MODORDERS-1162]. Order line encumbrance link not updated when re-encumber is false and order type not rolled over

### PR DESCRIPTION
**Pull request by BKadirkhodjaev**
## Purpose

- <https://folio-org.atlassian.net/browse/MODORDERS-1162>

## Approach

- Modify `buildOpenOrClosedOrderQueryByFundIdsAndTypes` to build a query without including any encumbrance settings
- Perform slight `OrderRolloverService` refactoring 

## Related issues

- <https://github.com/folio-org/mod-finance-storage/pull/424>
- <https://github.com/folio-org/folio-integration-tests/pull/1536>

### CODE REVIEW

File: src/main/java/org/folio/service/orders/OrderRolloverService.java [hunk 1]
```diff
@@ -144,15 +141,17 @@ private Future<Void> handleOrderRolloverError(Throwable t, LedgerFiscalYearRollo
...
-        List<Future<Void>> futures = new ArrayList<>();
+    var ctx =  requestContext.getContext();
+    return ctx
+      .<List<Future<Void>>>executeBlocking(promise -> {
+        var futures = new ArrayList<Future<Void>>();
         // process fundId chunks one by one
-        Semaphore semaphore = new Semaphore(1, requestContext.getContext().owner());
-        for (List<String> fundIds : fundIdChunks) {
+        var semaphore = new Semaphore(1, requestContext.getContext().owner());
+        for (var fundIds : fundIdChunks) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/orders/OrderRolloverService.java:154]: do you think that this formatting style is better then previous? for me it is the same, but let it be

Comment (BKadirkhodjaev) [reply src/main/java/org/folio/service/orders/OrderRolloverService.java:154]: I just saw that there are 2 styles here in this method i.e. reference types & var together, I think everything inside should be either one or the other to make it more readable

File: src/main/java/org/folio/service/orders/OrderRolloverService.java [hunk 2]
```diff
@@ -181,33 +180,29 @@ private void throwExceptionIfOneOfFailed(AsyncResult<Void> openOrdersResult, Asy
...
+    var query = buildOpenOrClosedOrderQueryByFundIdsAndTypes(chunkFundIds, workflowStatus, ledgerFYRollover);
     var totalRecordsFuture = purchaseOrderLineService.getOrderLineCollection(query, 0, 0, requestContext)
       .map(PoLineCollection::getTotalRecords);
-
     var ctx = requestContext.getContext();
-    return totalRecordsFuture.compose(totalRecords -> ctx.<List<Future<Void>>>executeBlocking(promise -> {
-        List<Future<Void>> futures = new ArrayList<>();
+    return totalRecordsFuture.compose(totalRecords ->
+      ctx.<List<Future<Void>>>executeBlocking(promise -> {
+        var futures = new ArrayList<Future<Void>>();
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/orders/OrderRolloverService.java:190]: if you think that previous code was not so redable we have util HelperUtils::executeWithSemaphores that encupsulate complexity, but we can remain as is

File: src/main/java/org/folio/service/orders/OrderRolloverService.java [hunk 3]
```diff
@@ -431,37 +423,38 @@ private String buildQueryEncumbrancesForRollover(List<String> polineIds, LedgerF
...
+    resultQuery.append(" sortBy metadata.createdDate");
+    logger.info("buildOpenOrClosedOrderQueryByFundIdsAndTypes:: Resulting PO line query: {}", resultQuery);
+    return resultQuery.toString();
+  }
 
-    resultQuery = resultQuery + sortByCreatedDate;
-    logger.debug("buildOpenOrClosedOrderQueryByFundIdsAndTypes :: resulting PO line query: {}", resultQuery);
-
-    return resultQuery;
+  private static String buildFundIdQuery(List<String> fundIds) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/orders/OrderRolloverService.java:445]: can be non static, static better to use to utility classes only

**Review by azizbekxm — APPROVED** [review]

**Pull request merged**